pixleesdk/src/main/java/com/pixlee/pixleesdk/PXLBaseAlbum.java

```diff
@@ -460,15 +460,52 @@ public void onFailure(Call<String> call, Throwable t) {
                 });
     }
 
-    /***
-     * Analytics methods
+    /**
+     * widgetVisible Analytics
+     * @param widgetType: PXLWidgetType enum class
+     * @return true: api called, false: cannot use this. please see the LogCat
      */
+    public boolean widgetVisible(PXLWidgetType widgetType) {
+        return widgetVisible(widgetType.getType());
+    }
 
+    /**
+     * widgetVisible Analytics
+     * @param widgetType: String
+     * @return true: api called, false: cannot use this. please see the LogCat
+     */
+    public boolean widgetVisible(String widgetType) {
+        return fireWidgetCall("events/widgetVisible", widgetType);
+    }
+
+    /**
+     * openedWidget Analytics
+     * @param widgetType: PXLWidgetType enum class
+     * @return true: api called, false: cannot use this. please see the LogCat
+     */
     public boolean openedWidget(PXLWidgetType widgetType) {
         return openedWidget(widgetType.getType());
     }
 
+    /**
+     * openedWidget Analytics
+     * @param widgetType: String
+     * @return true: api called, false: cannot use this. please see the LogCat
+     */
     public boolean openedWidget(String widgetType) {
+        return fireWidgetCall("events/openedWidget", widgetType);
+    }
+
+    private boolean fireWidgetCall(String requestPath, String widgetType){
+        if (album_id == null) {
+            Log.w(
+                    TAG,
+                    "missing album id. " +
```

Comment:
I created fireWidgetCall(String requestPath, String widgetType) in order to reduce redundancy of codes for making request parameters since the request parameters of WidgetVisible and OpenedWidget are the same. The only difference is the requestPath. fireWidgetCall reduce identical 50-line codes of one of the two APIs.